lua/iswap.lua

```diff
@@ -322,13 +315,21 @@ function M.iswap_with(direction, config)
     return
   end
 
-  local a_range, b_range = unpack(
-    internal.swap_nodes_and_return_new_ranges(cur_node, a, bufnr, config.move_cursor)
-  )
-
+  local a_range, b_range
+  if not a_idx or not imove then
```

Comment:
Shouldn't `if not a_idx` be an early return? I think it should be like:

```lua
if not a_idx then
  return
elseif not imove then
  normal swap
else
  move_nodes_to_index
end
```

Reply:
`a_idx` is not set in the cases of autoswap with 2 nodes and the `I*Left, I*Right` commands. Doing the swap is necessary but the index isn't needed for it 

This is actually I think I was wondering is that IMoveRight and IMoveLeft are actually logically the same as ISwapLeft and ISwapRight. I added those commands just for symmetry and also since iswap and imove share the same function underneath it just works. I think it's fine, seems unnecessary to explicitly prohibit it

Reply:
But in those cases, there is still a theoretical value that a_idx should be right? So if we move forward with only returning a_idx from ui.prompt, it should take the purpose of what `a` originally had, is my thinking

Reply:
Oh yeah if I do that change to ui.prompt then I'll change the conditional. Actually there is already a check just after ui.prompt so it would already have early returned